### PR TITLE
fix(shutdown): shutdown without sending signal

### PR DESCRIPTION
**Pull request by Matovidlo**

### CODE REVIEW

File: cadre.go [hunk 1]
```diff
@@ -44,27 +45,27 @@ type cadre struct {
 
 func (c *cadre) Start() error {
 	sigs := make(chan os.Signal, 1)
-	sigsDone := make(chan bool)
+	c.sigsDone = make(chan bool)
```
Comment (sveatlo) [inline cadre.go:48]: move to the top

File: cadre.go [hunk 2]
```diff
@@ -80,21 +81,28 @@ func (c *cadre) Start() error {
...
 	c.ctxCancel()
 	c.swg.Wait()
 
 	return nil
 }
 
+// This function shutdown the Start function that is waiting for sigsDone. The Start function initiates the context
+// cancelation and waits
+func (c *cadre) Shutdown() error {
+	c.sigsDone <- true
```
Comment (sveatlo) [inline cadre.go:102]: add close(c.sigsDone) to exit the signal-handling goroutine

**Review by sveatlo — CHANGES_REQUESTED** [review]

File: cadre.go (hunk 1 above)
Comment (sveatlo) [reply cadre.go:47]: sry, i meant that this should be moved to the cadre's construction, i.e.: Builder's `Build` function.

**Comment by Matovidlo** [comment]
:+1: Thanks for resolving the conflict

**Pull request merged**